daemon/state.go

```diff
@@ -65,8 +66,15 @@ func (d *Daemon) SyncState(dir string, clean bool) error {
 	for _, ep := range possibleEPs {
 		go func(ep *endpoint.Endpoint, epRestored, epRegenerated chan<- bool) {
 			scopedLog := log.WithField(logfields.EndpointID, ep.ID)
-			if clean && !containerd.IsRunning(ep) {
+			skipRestore := false
+			if _, err := netlink.LinkByName(ep.IfName); err != nil {
```

Comment:
Can you replace this with `if err == netlink.LinkNotFoundError`?

Reply:
In a similar vein, could we gate the whole section by clean, then do the two checks (with early returns) and have the `d.deleteEndpointQuiet` bit be the default? I'm not sure if we want to log the missing Link Info every time it happens, but the "No workload" message is now logged in more cases than before (because it isn't gated by `clean`). Alternatively, a switch might work well here: gate by `clean`, two switch cases for each early return, and the default for the deletion.

Reply:
Oh, nevermind. I see now that we also want to skip restore if the device isn't there, irrespective of clean.

Reply:
@aanm will do, and if the error is something else, log the error.

@raybejjani In practice `clean` is always true. A follow up cleanup can get rid of the `clean` variable, but I wasn't sure if this might need backporting so I wanted to keep it minimal.

Reply:
@aanm there's no instantiation of that error type, so I'm walking back that change.